base/traits.jl

```diff
@@ -2,29 +2,29 @@
 
 ## numeric/object traits
 # trait for objects that have an ordering
-abstract type TypeOrder end
-struct HasOrder <: TypeOrder end
-struct Unordered <: TypeOrder end
+abstract type OrderStyle end
+struct HasOrder <: OrderStyle end
```

Comment:
As suggested at 25440, maybe also rename `HasOrder` to `Ordered` while you're at it?